mesonbuild/environment.py

```diff
@@ -1289,6 +1283,11 @@ def get_root(self):
     def get_sys_root(self):
         return self.properties.get('sys_root', None)
 
+    def __eq__(self, other):
+        if self.__class__ is other.__class__:
```

Comment:
hmmmm, I'd prefer if this at least handled subclasses, `if isinstance(other, type(self))`. We really shouldn't be directly mucking with the dunders anyway.